app/src/main/java/org/dhis2/data/service/ServiceModule.java

```diff
@@ -23,7 +25,13 @@ NotificationManager notificationManager(@NonNull Context context) {
 
     @Provides
     @PerService
-    SyncPresenter syncPresenter(@NonNull D2 d2) {
-        return new SyncPresenterImpl(d2);
+    SyncPresenter syncPresenter(
+            @NonNull Context context,
+            @NonNull D2 d2,
+            @NonNull PreferenceProvider preferences
+    ) {
+        WorkManager workManager = WorkManager.getInstance(context);
```

Comment:
We could add the WorkManager in the application module so that we can inject it.

Reply:
You are right, the WorkManager is already added in the appModule. There is no need to do this anymore. Will remove